### PR TITLE
Allow passed in plugins/presets to override the defaults

### PR DESCRIPTION
**Pull request by matthewp**
The way plugins/presets should be processed is that if you define your
own you are essentially on your own (aside from the module format, which
		we have to set ourselves).

Previously we were always included some default plugins that made code
always compile to ES5. This fixes that issue.

### CODE REVIEW

File: lib/babel_compile.js [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
-	opts.plugins = opts.plugins || [];
+	if(empty(opts.presets) && empty(opts.plugins)) {
+		opts.presets = defaultPresets.concat(opts.presets || []);
+		opts.plugins = opts.plugins || [];
+	} else {
+		opts.presets = opts.presets || [];
+		opts.plugins = opts.plugins || [];
+
+		var formatPlugin;
+		if(format === "commonjs") {
```
Comment (m-mujica) [inline lib/babel_compile.js]: where is format set to `commonjs` in steal-tools/export?

File: lib/babel_compile.js [hunk 2]
```diff
@@ -1,21 +1,38 @@
...
+			formatPlugin = "transform-es2015-modules-commonjs";
+		} else if(format === "amd") {
+			formatPlugin = "transform-es2015-modules-amd";
+		}
+		if(formatPlugin && opts.plugins.indexOf(formatPlugin) === -1) {
```
Comment (m-mujica) [inline lib/babel_compile.js]: Nit: `!options.plugins.includes(formatPlugin)`

File: lib/babel_compile.js [hunk 3]
```diff
@@ -25,7 +42,14 @@ module.exports = function(source, compileOptions, options){
...
+		opts.sourceRoot = options.sourceRoot;
 	}
 
-	return babel.transform(source, opts);
+	var result = babel.transform(source, opts);
+	toEstree(result.ast, source);
+	return result;
 };
+
+function empty(arr) {
```
Comment (m-mujica) [inline lib/babel_compile.js]: https://lodash.com/docs/4.17.5#isEmpty .... just saying...

**Comment by matthewp** [comment]
This isn't going to be used anymore, instead went with much simpler changes in #135

**Pull request closed without merging**